a3/src/main/java/org/cloudname/a3/A3Client.java

```diff
@@ -116,6 +108,39 @@ public static A3Client newMemoryOnlyClient(Reader in) throws IOException {
         return new A3Client(MemoryStorage.fromReader(in));
     }
 
+    public GetUserResult getUserByProperty(final String key, final String value) {
+        final UserDB userDB;
+        try {
+            userDB = getUserDb();
+        } catch (Exception e) {
+            return new GetUserResult(GetUserResult.State.INTERNAL_ERROR,
+                    "Unable to get user database",
+                    e);
+        }
+        final Set<User> users = new HashSet<User>();
+        for (String username : userDB.getUserNames()) {
+            final User user = userDB.getUser(username);
+            final Map<String, String> properties = user.getProperties();
+            if (properties != null && value.equals(properties.get(key))) {
```

Comment:
The problem is that the current implementation will not actually work. I think since it just looks up any property by name, it is generic enough to warrant inclusion in cloudname, while we in sylfide will add a more specific getUserByClientId which is locked to the actual property to lookup on.